ext/date/php_date.c

```diff
@@ -4124,28 +4124,16 @@ static int date_period_initialize(timelib_time **st, timelib_time **et, timelib_
 	return retval;
 } /* }}} */
 
-/* {{{ Creates new DatePeriod object. */
-PHP_METHOD(DatePeriod, __construct)
-{
-	php_period_obj   *dpobj;
-	php_date_obj     *dateobj;
-	zval *start, *end = NULL, *interval;
-	zend_long  recurrences = 0, options = 0;
-	char *isostr = NULL;
-	size_t   isostr_len = 0;
+static void construct_date_period(
+	INTERNAL_FUNCTION_PARAMETERS, zval *object, zval *start, zval *end, zval *interval, zend_long recurrences,
+	zend_long options, char *isostr, size_t isostr_len
```

Comment:
I would prefer it if this initialisation function gets split as well. The old ``__construct`` should use the new (3) initialisation methods, which also makes it easier to clean up in for example PHP 9.0. It can also mean that error and bounds checking (for example, recurrences can't be negative) to the right and logical position.